dev/ColorPicker/ColorPicker.cpp

```diff
@@ -428,6 +432,18 @@ void ColorPicker::OnColorSpectrumComponentsChanged(winrt::DependencyPropertyChan
     SetThirdDimensionSliderChannel();
 }
 
+void ColorPicker::OnOrientationChanged(winrt::DependencyPropertyChangedEventArgs const& args)
+{
+    if (unbox_value<winrt::Orientation>(args.NewValue()) == winrt::Orientation::Horizontal)
+    {
+        winrt::VisualStateManager::GoToState(*this, L"Horizontal", true);
+    }
+    else
+    {
+        winrt::VisualStateManager::GoToState(*this, L"Vertical", true);
+    }
```

Comment:
I would move this bit into UpdateVisualState and simply call UpdateVisualState. This keeps all the VisualState transitions consistently in one place.

Reply:
Sure thing, updated the code now. Good idea!